console/frontend/src/main/frontend/src/angularjs/app/views/webservices/webservices.component.ts

```diff
@@ -0,0 +1,29 @@
+import { appModule } from "../../app.module";
+import { ApiService } from "../../services/api.service";
+import { MiscService } from "../../services/misc.service";
+
+class WebservicesController {
+  rootURL: string = "";
+
+  constructor(
+    private Api: ApiService,
+    private Misc: MiscService
+  ) { };
+
+  $onInit() {
+    this.rootURL = this.Misc.getServerPath();
+
+    this.Api.Get("webservices", (data) => {
+      Object.assign(this, data);
+    });
+  };
+
+  compileURL(apiListener: any) {
+    return this.rootURL + "iaf/api/webservices/openapi.json?uri=" + encodeURI(apiListener.uriPattern);
+  };
+};
+
+appModule.component('webservices', {
+  controller: ['Api', 'Misc', WebservicesController],
+  templateUrl: 'js/app/views/webservices/webservices.component.html'
```

Comment:
Same for this one

Reply:
This is the angularjs file that is now converted to ts and later copied over to the angular version of this component, also this is a template url thats not dynamic in any way shape or form, this wont ever change anymore since its only necessary to load the angularjs version of this component if needed to and will be removed once everything has moved over to angular

Reply:
Alright, thanks for clarification.